lib/ansible/utils/display.py

```diff
@@ -48,6 +49,50 @@
     # Python 3, we already have raw_input
     pass
 
+# Set the locale to the users default setting
+locale.setlocale(locale.LC_ALL, '')
+
+_LIBC = ctypes.cdll.LoadLibrary(ctypes.util.find_library('c'))
+# Set argtypes, to avoid segfault if the wrong type is provided,
+# restype is assumed to be c_int
+_LIBC.wcwidth.argtypes = (ctypes.c_wchar,)
+_LIBC.wcswidth.argtypes = (ctypes.c_wchar_p, ctypes.c_int)
+# Max for c_int
+_MAX_INT = 2 ** (ctypes.sizeof(ctypes.c_int) * 8 - 1) - 1
+
+
+def get_text_width(text):
+    """Function that utilizes ``wcswidth`` or ``wcwidth`` to determine the
+    number of columns used to display a text string.
+
+    We try first with ``wcswidth``, and fallback to iterating each
+    character and using wcwidth individually, falling back to a value of 0
+    for non-printable wide characters
+
+    On Py2, this depends on ``locale.setlocale(locale.LC_ALL, '')``,
+    that in the case of Ansible is done in ``bin/ansible``
+    """
+    text = to_text(text)
```

Comment:
Since Ansible subscribes to unicode sandwich, this should be the callers' (actually, where the data enters the system) responsibility.